sessionctx/variable/session.go

```diff
@@ -1866,6 +1883,25 @@ func (s *SessionVars) DecodeSessionStates(ctx context.Context, sessionStates *se
 		s.UserVarTypes[name] = userVarType.Clone()
 	}
 	s.UsersLock.Unlock()
+
+	// Decode other session contexts.
+	s.preparedStmtID = sessionStates.PreparedStmtID
+	s.Status = sessionStates.Status
+	s.CurrentDB = sessionStates.CurrentDB
+	s.LastTxnInfo = sessionStates.LastTxnInfo
+	if sessionStates.LastQueryInfo != nil {
+		s.LastQueryInfo = *sessionStates.LastQueryInfo
+	}
```

Comment:
I think clone is better than `&s.LastQueryInfo`. Could we use `sessionStates.StartTS != 0` instead of  `sessionStates.LastQueryInfo != nil` to judge if we should fill `LastQueryInfo` from `sessionStates`, thus eliminate escaping `LastQueryInfo`?